packages/client/src/comp/states/BaseContainer.ts

```diff
@@ -63,13 +64,24 @@ const useFields = () => {
     mailing: { valid: true, value: '' },
   })
 
-   /**
-   * Detects the content of all inputs needed for checking voter registration.
-   * When possible, this function will contact the server, updating `registrationStatus`
-   * when finished.
-   */
+  const nameParts: NameParts = {
+    first: fields.firstName.value,
+    middle: fields.middleName.value ? fields.middleName.value : undefined,
+    last: fields.lastName.value,
+    suffix: fields.suffix.value ? fields.suffix.value : undefined,
+  }
+
   const updateField = (id: InputId, value: string) => {
     _updateValid({ ...fields, [id]: { valid: isInputValid(id, value), value } })
+    if (id.indexOf('Name') !== -1 || id === 'suffix') {
```

Comment:
Let's write this out explicitly using swtich case.  This kind of code is pretty hacky.

-- Sorry, I just noticed this one.